test/functional/api/v2/compute_resources_controller_test.rb

```diff
@@ -289,4 +289,18 @@ def teardown
     cr.reload
     assert_nil cr.attrs[:setpw]
   end
+
+  test "should not update display_type for non-Libvirt compute resource" do
+    cr = compute_resources(:openstack)
+    put :update, { :id => cr.id, :compute_resource => { :display_type => 'SPICE' } }
+    cr.reload
+    assert_nil cr.attrs[:display]
+  end
+
+  test "should update display type for Libvirt compute resource" do
```

Comment:
This would be much better and more efficient as a unit test, if you're going to extend ComputeResource or want to test the Libvirt model.  I don't think testing the API's valuable here.